tutoraspects/plugin.py

```diff
@@ -227,6 +227,38 @@
     </users>
             """,
         ),
+        # Similar to CLICKHOUSE_EXTRA_XML_CONFIG, this will be read from a
+        # mount on /etc/clickhouse-server/config.d/named_collections.xml
+        # on startup. Named collections allow configuration to talk to external
+        # or internal services using stored credentials. We use them to create
+        # dictionaries on local queries, but credentials are still required.
+        # By placing them in named collections we can change the ClickHouse
+        # admin password without having to disrupt the dictionary caching.
+        # https://clickhouse.com/docs/en/operations/named-collections
+        #
+        # If you are creating your own dictionaries, you may need to add more
+        # named collections here first.
+        (
+          "CLICKHOUSE_EXTRA_NAMED_COLLECTION_XML_CONFIG",
+          """
+        <local_ch_xapi>
+            <host overridable="false">localhost</host>
+            <port overridable="false">{{ CLICKHOUSE_INTERNAL_NATIVE_PORT }}</port>
+            <database overridable="false">{{ ASPECTS_XAPI_DATABASE }}</database>
+            <user overridable="false">{{ ASPECTS_CLICKHOUSE_REPORT_USER }}</user>
+            <password overridable="false">{{ ASPECTS_CLICKHOUSE_REPORT_PASSWORD }}</password>
+            <secure overridable="false">{% if CLICKHOUSE_SECURE_CONNECTION %}1{% else %}0{% endif %}</secure>
+        </local_ch_xapi>
+        <local_ch_event_sink>
+            <host overridable="false">localhost</host>
+            <port overridable="false">{{ CLICKHOUSE_INTERNAL_NATIVE_PORT }}</port>
+            <database overridable="false">{{ ASPECTS_EVENT_SINK_DATABASE }}</database>
+            <user overridable="false">{{ ASPECTS_CLICKHOUSE_REPORT_USER }}</user>
+            <password overridable="false">{{ ASPECTS_CLICKHOUSE_REPORT_PASSWORD }}</password>
+            <secure overridable="false">{% if CLICKHOUSE_SECURE_CONNECTION %}1{% else %}0{% endif %}</secure>
+        </local_ch_event_sink>
+          """
+        ),
```

Comment:
This is a too large to be a setting, can we use a patch for it and template it?

Also, as Clickhouse resolves any conflicts we can just override any value here using the patch

Reply:
I did this to match the other ClickHouse settings (`CLICKHOUSE_EXTRA_XML_CONFIG`, `CLICKHOUSE_EXTRA_USERS_XML_CONFIG`) so that they're not being configured in two different ways. I can make all of those patches, but I think that requires people to create a plugin to override them instead of just changing a config value? Just making sure since we would probably need to document the two types of settings clearly to say how to override the different settings.

Reply:
It's also worth noting that `CLICKHOUSE_EXTRA_USERS_XML_CONFIG` in the current configuration doesn't work on k8s as well. I wasn't able to solve that problem as part of this.

Reply:
That's a too large configuration to be managed as a setting. I would prefer the patch